### PR TITLE
New reader for .asc files

### PR DESCRIPTION
**Pull request by phuang26**
Asc files will be transmitted as bundled zip archives. The reader unzips the files and reads them as collective

### CODE REVIEW

**Comment by jochenklar** [comment]
Is this a duplicate of #41? I am a bit confused.

File: converter_app/readers/asc_zip.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        formatedResults = self.formatResultsChemotion(all_results)
+        table['metadata'] = formatedResults['metadata']
+        col_names = list(formatedResults['data'].keys())
+        table['columns'] = [{
+            'key': str(idx),
+            'name': '{}'.format(value)
+        } for idx, value in enumerate(col_names)]
+        for idx in range(len(formatedResults['data'][col_names[0]])):
+            table['rows'].append([formatedResults['data'][name][idx] for name in col_names])
+        return tables
```
Comment (jochenklar) [inline converter_app/readers/asc_zip.py:143]: Here the column/rows count is missing:

```
table['metadata']['rows'] = str(len(table['rows']))
table['metadata']['columns'] = str(len(table['columns']))
```

File: converter_app/readers/asc_zip.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        metadata["Samplename"] = results[0]["Samplename"]
+        # DLS device = Device Info --> options
+        metadata["Device Info"] = results[0]["Device Info"].split("/")[0]
+        # TODO does this make sense?
+        # TODO sample description = sample memo? free-fill in elab?
+        # if results[0]['SampleMemo']:
+
+        # solvent = nothing, needs to be filled in manually for dls (?)
+        # sample volume [ml] = nothing, needs to be filled in manually for dls (?)
+        metadata["wavelength [nm]"] = results[0]["Wavelength [nm]"]
```
Comment (jochenklar) [inline converter_app/readers/asc_zip.py:67]: should be casted to `str()` here:

```
metadata["wavelength [nm]"] = str(results[0]["Wavelength [nm]"])
```

File: converter_app/readers/asc_zip.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        # hydrodynamic radius [nm] --> Hydrodynamic Radius 2. order fit [nm]
+        data["hydrodynamic radius [nm]"] = AscHelper.listValues("Hydrodynamic Radius 2. order fit [nm]", results)
+
+        for (key, val) in data.items():
+            metadata[key] = AscHelper.strVals(val)
+
+
+
+        # duration [s] = Duration [s]
+        metadata["duration [s]"] = results[0]["Duration [s]"]
```
Comment (jochenklar) [inline converter_app/readers/asc_zip.py:112]: Should be casted to `str()` here:

```
metadata["duration [s]"] = str(results[0]["Duration [s]"])
```

**Review by jochenklar — CHANGES_REQUESTED** [review]
For some reason, the data arrays end up in the table metadata, is that on purpose?

image

**Comment by jochenklar** [comment]
@phuang26 @StarmanMartin Ok, I added some remarks. But don't forget that the code is already on master, so I guess it would be best to just fix those things in master or yet another feature branch.

**Comment by StarmanMartin** [comment]
Fixed all commends:
[Pull request fixes](https://github.com/ComPlat/chemotion-converter-app/pull/42/commits/a32d08f6766e664ecf5dc37ca60914dc71dfd037)

**Pull request merged**